scipy/optimize/tests/test_optimize.py

```diff
@@ -1096,6 +1096,30 @@ def hess(x):
                                         options=dict(maxiter=20))
                 assert_equal(sol.success, False)
 
+    @pytest.mark.parametrize('method', ['nelder-mead', 'cg', 'bfgs',
+                                        'l-bfgs-b', 'tnc',
+                                        'cobyla', 'slsqp', 'trust-constr',
+                                        'dogleg', 'trust-ncg', 'trust-exact',
+                                        'trust-krylov'])
+    def test_duplicate_evaluations(self, method):
+        # check that there are no duplicate evaluations for any methods
+        jac = hess = None
+        if method in ('newton-cg', 'trust-krylov', 'trust-exact',
+                      'trust-ncg', 'dogleg'):
+            jac = self.grad
+        if method in ('trust-krylov', 'trust-exact', 'trust-ncg',
+                      'dogleg'):
+            hess = self.hess
+
+        x = self.startparams
+        self.trace = []
```

Comment:
Is this needed? Is `setup_method` called before each parameter value?

Reply:
probably not.